ui/core/src/model/calculations.ts

```diff
@@ -55,56 +56,151 @@ export const CALCULATIONS_CONFIG: Readonly<Record<CalculationType, CalculationCo
     label: 'Sum',
     description: 'The sum of all values',
   },
+  Min: {
+    label: 'Min',
+    description: 'Minimum value',
+  },
+  Max: {
+    label: 'Max',
+    description: 'Maximum value',
+  },
 } as const;
 
 export const DEFAULT_CALCULATION: CalculationType = 'Sum';
 
 type CalculationValue = number | null | undefined;
 
+/**
+ * Calculate a multiple values for a set of time series data.
+ *
+ * @param values - Array of time series data.
+ * @param includeCalculations - Array of calculations to include.
+ */
+export function getCalculations<IncludeCalcs extends CalculationType[]>(
+  values: TimeSeriesValueTuple[],
+  includeCalculations: IncludeCalcs
+): Record<
+  // This extract combined with the generics above keeps the key of the returned
+  // record to *just* the specified calculations.
+  Extract<CalculationType, IncludeCalcs[number]>,
+  CalculationValue
+> {
+  const calculations = includeCalculations.reduce((initResult, calculation) => {
+    initResult[calculation] = undefined;
+    return initResult;
+  }, {} as Record<string, CalculationValue>);
+
+  // We save these values as separate values instead of directly setting them
+  // in the calculations because they are needed by multiple calculations.
+  let nonNullCount = 0;
+  let sum = 0;
+
+  // We use this large function capable of performing one or more calculations
+  // in a single iteration of the data to minimize the performance impact of
+  // generating multiple calculations for large timeseries values. This is
+  // less optimized for certain single calculations when done in isolation (e.g.
+  // `Last`), but will be more performant in the more expensive cases where
+  // multiple values are being used (e.g. table legend).
```

Comment:
This is a very long way of explaining that it's better to iterate once if you're using really large data sets, which we can have with some time series.

Reply:
For context, `Last` would be cheaper as a separate calc because you can just do `values[values.length - 1]` and completely avoid iterating all the values. We could put a check for that in here, but I think it would overly complicate the code for a tiny optimization (as opposed to the potentially larger optimization around calculating multiple values that do require iterating).